### PR TITLE
Misc Extended Query Tag Fixes

### PR DESCRIPTION
**Pull request by wsugarman**
## Description
- Ensure user has read access for fetching operation status
- Ignore DICOM instances that have been deleted between fetching the batch and performing the re-index
- Add non-root user to docker image for functions
- Update enum serialization to strictly use names
- Update swagger doc to include new APIs and use enum names instead of numbers

## Related issues
[AB#87003](https://microsofthealth.visualstudio.com/Health/_workitems/edit/87003)

## Testing
Test pipelines

### CODE REVIEW

File: src/Microsoft.Health.Dicom.Core/Features/Operations/OperationStatusHandler.cs [hunk 1]
```diff
@@ -48,6 +54,12 @@ public async Task<OperationStatusResponse> Handle(OperationStatusRequest request
         {
             // TODO: Check for data action
```
Comment (wsugarman) [inline src/Microsoft.Health.Dicom.Core/Features/Operations/OperationStatusHandler.cs:55]: Will delete

**Review by pengchen0692 — APPROVED** [review]

**Review by alishahahmed — APPROVED** [review]

**Review by pengchen0692 — APPROVED** [review]

File: src/Microsoft.Health.Dicom.Api/Features/Swagger/ReflectionTypeFilter.cs [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+using Microsoft.OpenApi.Models;
+using Swashbuckle.AspNetCore.SwaggerGen;
+
+namespace Microsoft.Health.Dicom.Api.Features.Swagger
+{
+    internal class ReflectionTypeFilter : IDocumentFilter
+    {
+        private static readonly HashSet<Type> CollectionTypes = typeof(List<>).Assembly
+            .ExportedTypes
+            .Where(x => x.Namespace == "System.Collections.Generic")
```
[the remark this answers is not in the document]
Comment (wsugarman) [reply src/Microsoft.Health.Dicom.Api/Features/Swagger/ReflectionTypeFilter.cs:19]: I actually tried this and `nameof(System.Collections.Generic)` returns `"Generic"` lol

File: src/Microsoft.Health.Dicom.Api/Features/Swagger/ReflectionTypeFilter.cs [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using Microsoft.OpenApi.Models;
+using Swashbuckle.AspNetCore.SwaggerGen;
+
+namespace Microsoft.Health.Dicom.Api.Features.Swagger
+{
+    internal class ReflectionTypeFilter : IDocumentFilter
```
[the remark this answers is not in the document]
Comment (wsugarman) [reply src/Microsoft.Health.Dicom.Api/Features/Swagger/ReflectionTypeFilter.cs:15]: Yeah, I'll add some comments

File: src/Microsoft.Health.Dicom.Api/Features/Filters/BodyModelStateValidatorAttribute.cs [hunk 4]
```diff
@@ -17,7 +17,7 @@ public override void OnActionExecuting(ActionExecutingContext context)
             EnsureArg.IsNotNull(context, nameof(context));
             if (!context.ModelState.IsValid)
             {
-                var error = context.ModelState.First(x => x.Value.ValidationState == AspNetCore.Mvc.ModelBinding.ModelValidationState.Invalid);
```
[the remark this answers is not in the document]
Comment (wsugarman) [reply src/Microsoft.Health.Dicom.Api/Features/Filters/BodyModelStateValidatorAttribute.cs:20]: It seems the first error may be less specific, like in the case that a JSON member fails to bind for a body parameter. The 1st error will indicate a problem with the body, while the 2nd error will be more specific.

**Pull request merged**